app/Http/Controllers/v1/Actions/StoreUserEnrollmentController.php

```diff
@@ -14,41 +14,35 @@ class StoreUserEnrollmentController extends Controller
     /**
      * Handle the incoming request.
      */
-    public function __invoke(AdminRequest $request, int $providerId, string $userId)
+    public function __invoke(AdminRequest $request)
     {
-        $canvasService = CanvasServices::byProviderId($providerId);
-        $canvasEnrollments = $canvasService->listEnrollmentsForUser($userId);
+        $canvasService = CanvasServices::byProviderId($request->provider_platform_id);
+        $canvasEnrollments = $canvasService->listEnrollmentsForUser($request->user_id);
         $enrollmentCollection = collect();
         foreach ($canvasEnrollments as $enrollment) {
-            if ($course = Course::where('provider_resource_id', $enrollment->course_id)->firstOrFail()) {
+            if ($course = Course::where('external_resource_id', $enrollment->course_id)->firstOrFail()) {
                 $request->merge([
-                    'user_id' => $userId,
+                    'user_id' => $request->user_id,
                     'course_id' => $course->id,
-                    'provider_user_id' => $enrollment->user_id,
-                    'provider_course_id' => $enrollment->course_id,
-                    'provider_platform_id' => $providerId,
-                    'provider_enrollment_id' => $enrollment->id,
-                    // 'provider_course_name' => $enrollment->sis_course_id,
+                    'external_enrollment_id' => $enrollment->id,
                     'enrollment_state' => $enrollment->enrollment_state,
-                    'links' => [],
```

Comment:
Agreed, `links` was a question mark in my mind too. Vague catchall that didn't correspond to any canvas enrollment data.